[Feng Zhunyi] ip

Leofeng10 opened the pull request:
# Duke Pro
>“Your mind is for having ideas, not holding them.” – David Allen ([source](http://github.com))
## DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~  *SUPER FAST* to use
## All you need to do is,
1. download it from ([here](http://github.com)).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**
## Features
- [x] Managing tasks
- [ ] Reminders (coming soon)
## If you Java programmer, you can use it to practice Java too. Here's the main method:
```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Level-1 commitment
Basic Info for ip: Greeting, echo, exit

Leofeng10 committed:
level-2

Leofeng10 committed:
Level-3

Leofeng10 committed:
Level-4

Leofeng10 committed:
A-TextUiTesting

Leofeng10 committed:
Level-5

Leofeng10 committed:
Level-6

Leofeng10 committed:
Level-6 bug fixed

Leofeng10 committed:
Enumeration

Leofeng10 committed:
bug fixed

Leofeng10 committed:
bug fixed

Leofeng10 committed:
Level-7

Leofeng10 committed:
Revert "Level-7"

This reverts commit 8ea8fc469a20a6d9cf6f1b2dca0fa70d0dd05cf7.

Leofeng10 committed:
Revert "Level-7"

This reverts commit 8ea8fc469a20a6d9cf6f1b2dca0fa70d0dd05cf7.

Leofeng10 committed:
bug fixed

Leofeng10 committed:
Merge branch 'test'

Leofeng10 committed:
Level-7 bug fixed

Leofeng10 committed:
Merge branch 'branch-Level-7'

Leofeng10 committed:
bug fixed

Leofeng10 committed:
level-8

Leofeng10 committed:
bug fixed

Leofeng10 committed:
A-MoreOOP

Leofeng10 committed:
A-Packages

Leofeng10 committed:
A-JUnit

Leofeng10 committed:
A-JavaDoc

Leofeng10 committed:
A-CodingStandard

Leofeng10 committed:
UPDATE

Leofeng10 committed:
update

Leofeng10 committed:
update

Leofeng10 committed:
Level-9

Leofeng10 commented:
# Duke Pro
>“Your mind is for having ideas, not holding them.” – David Allen ([source](http://github.com))
## DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~  *SUPER FAST* to use
## All you need to do is,
1. download it from ([here](http://github.com)).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**
## Features
- [x] Managing tasks
- [ ] Reminders (coming soon)
## If you Java programmer, you can use it to practice Java too. Here's the main method:
```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Leofeng10 commented on src/duke/core/UI.java:
```diff
@@ -0,0 +1,28 @@
+/**
+ * Message and ui for duke
+ */
+public class UI {
```
test

Leofeng10 committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-Gradle

Leofeng10 committed:
A-Gradle

Leofeng10 committed:
check style

Chen-Yiyang commented on src/main/java/duke/core/Duke.java:
```diff
@@ -0,0 +1,339 @@
+    }
+
+    /**
+     * start the program
```
Should this start with a word 'Starts' instead of 'start'? I have noticed similar points in your javadoc elsewhere as well.

Leofeng10 replied:
I see! Thanks.

Chen-Yiyang commented on src/main/java/duke/core/Duke.java:
```diff
@@ -0,0 +1,339 @@
+
+
+            switch(keyword[0]) {
+            case FIND:
```
I like how you layout your switch structure nicely and clearly. Readability enhanced!

Chen-Yiyang commented on src/main/java/duke/core/TaskList.java:
```diff
@@ -0,0 +1,17 @@
+package duke.core;
+
+import java.util.ArrayList;
```
I like how you put a space between the import statements to indicate that they belong to different groups (Java API vs. other package files)!

Chen-Yiyang commented on src/main/java/duke/databse/Database.java:
```diff
@@ -0,0 +1,215 @@
+
+    /**
+     * update data to file
+     * @param task
```
Maybe you want to add in more comments explaining each of these parameters?

Leofeng10 replied:
Noted! That's true

Chen-Yiyang commented on src/main/java/duke/core/UI.java:
```diff
@@ -0,0 +1,28 @@
+    public final String bye_message = line + "\n" + indentation + "Bye. Hope to see you again soon!" + "\n" + line;
+    public final String no_task_message = line + "\n" + indentation + "Sorry, you do not have this task" + "\n" + line;
+    public final String task_num_message = line + "\n" + indentation + "Sorry, please enter a task number bigger than 0" + "\n" + line;
+    public final String lack_content_message = line + "\n" + indentation + "OOPS!!! The description or time of a task cannot be empty." + "\n" + line;
```
Maybe you can consider using line wrapping since it is quite long?

Leofeng10 replied:
yes, I created a new class for all of these strings but the class was not added in this commit.

Leofeng10 replied:
That's true, they are too long. But actually my intension of having this class is just to have all the strings I need for outputs. So I just put them in a line.

Chen-Yiyang reviewed (commented):
Great work done. Nice code structure. Very readable code! 👍

Leofeng10 committed:
gui

luayida99 commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,51 @@
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+import duke.core.*;
```
Should the import statements be more explicitly stated instead of using *?

Leofeng10 replied:
That's true

luayida99 commented on src/main/java/duke/core/UI.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class UI {
+
+    public final String indentation = "       ";
+    public final String line = indentation + "---------------------------------------------------------------";
+    public final String greeting = indentation + "Hello! I'm Duke developed by Feng Zhunyi.\n" + indentation + "What can I do for you?\n";
+    public final String logo = indentation + " ____        _        \n"
+            + indentation + "|  _ \\ _   _| | _____ \n"
+            + indentation + "| | | | | | | |/ / _ \\\n"
+            + indentation + "| |_| | |_| |   <  __/\n"
+            + indentation + "|____/ \\__,_|_|\\_\\___|\n";
+    public final String done_message = indentation + "Nice! I've marked this task as done:";
+    public final String remove_message = indentation + "Noted. I've removed this task:";
+    public final String bye_message = line + "\n" + indentation + "Bye. Hope to see you again soon!" + "\n" + line;
+    public final String no_task_message = line + "\n" + indentation + "Sorry, you do not have this task" + "\n" + line;
+    public final String task_num_message = line + "\n" + indentation + "Sorry, please enter a task number bigger than 0" + "\n" + line;
+    public final String lack_content_message = line + "\n" + indentation + "OOPS!!! The description or time of a task cannot be empty." + "\n" + line;
+    public final String index_message = line + "\n" + indentation + "please follow the format of adding task" + "\n" + line;
+    public final String unknown_message = line + "\n" + indentation + "OOPS!!! I'm sorry, but I don't know what that means :-(" + "\n" + line;
+    public final String added_message = line + "\n" + indentation + "Got it. I've added this task:" + "\n";
+    public final String reading_file_message = indentation + "Reading your tasks from record : )";
+    public final String creating_file_message = indentation + "Creating a file to store your tasks : )";
+    public final String wrong_message = index_message + "OOPS!! Something wrong!";
```
Could consider using line wrapping here? Strings assigned here are quite long.

Leofeng10 replied:
yes, I created a new class for all of these strings but the class was not added in this commit.

Leofeng10 replied:
That's true, they are too long. But actually my intension of having this class is just to have all the strings I need for outputs. So I just put them in a line.

luayida99 commented on src/main/java/duke/databse/Database.java:
```diff
@@ -0,0 +1,215 @@
+        } else {
+            isDone = true;
+        }
+        switch (s[0]) {
```
Could consider assigning s[0] to a variable before using it in this switch statement? Understand your intention but would be more readable if the variable to compare is a bit more explicit.

luayida99 commented on src/main/java/duke/databse/Database.java:
```diff
@@ -0,0 +1,215 @@
+    }
+
+    /**
+     * write data to file
+     * @param todo
```
Consider detailing what the parameters do in javadocs? Seems like many variables are not elaborated on in your javadocs

Li-Lehao replied:
Maybe this header comment should be `Writes data to file.` instead? For the parameter `todo`, we need to add descriptions also.

luayida99 commented on src/main/java/duke/databse/Database.java:
```diff
@@ -0,0 +1,215 @@
+     */
+    public void writeToDatabase(Task todo) {
+        try {
+            BufferedWriter out = new BufferedWriter(
+                    new FileWriter("todoList.txt", true));
+            out.write(todo.toString());
```
Good use of abstraction here, makes it easy to understand what is going on!

luayida99 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,84 @@
+
+    /**
+     * get time
+     * @return
```
Some javadocs seem to be missing, consider filling them in when you have time?

Leofeng10 replied:
thanks for reminding

luayida99 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,84 @@
+     * @param keyword
+     * @return
+     */
+    public boolean searchKeyword(String keyword) {
```
Could make this sound more boolean? 
```suggestion
    public boolean hasKeyword(String keyword) {
```

luayida99 reviewed (commented):
Good work overall! No major problems with quality and code is generally readable 👍

Leofeng10 committed:
A-Varargs

hmanhduc2k commented on src/duke/core/Duke.java:
```diff
@@ -22,11 +22,18 @@
+                result.add(task);
+            }
+        }
+        return result;
```
Consider adding JavaDoc comment on all of your implementations. It will help make it easier for people to understand what your code does, even if it seems obvious.

hmanhduc2k commented on src/duke/core/Duke.java:
```diff
@@ -57,6 +65,37 @@ public void run() {
+                            } else {
+                                s += "[X] " + s2;
+                                System.out.println(s);
+                            }
```
Your code is too deeply-nested, with many layers of conditional branches. and the variable naming are a bit non-comprehensible. Do consider breaking down your code and make variables more obvious for readers to understand what it is about

hmanhduc2k commented on src/duke/core/Duke.java:
```diff
@@ -57,6 +65,37 @@ public void run() {
 
             switch(keyword[0]) {
+                case FIND:
+                    ArrayList<Task> result = searchTask(task, keyword[1]);
```
Speaking of which, it would be better for you to reduce the code size of your methods and possibly implement a different class, so that your code for the FIND command will be more readable.

hmanhduc2k commented on src/duke/task/Task.java:
```diff
@@ -18,6 +18,16 @@ public Task(String name, boolean isDone, int index) {
+            }
+        }
+        return false;
+    }
```
Consider adding JavaDoc to all these methods for readability purposes. Speaking of which, it might be necessary to make them inherit or implement a certain class/interface so that you would not have to implement the searchKeyword too many time.

hmanhduc2k reviewed (commented):
Overall, your code is quite good, you show effort in following the module standard of coding, and it would be better if you can change minor things to match the standard given even more. Do consider adding JavaDoc for better understanding.

Li-Lehao commented on src/main/java/duke/core/UI.java:
```diff
@@ -0,0 +1,28 @@
+    public final String bye_message = line + "\n" + indentation + "Bye. Hope to see you again soon!" + "\n" + line;
+    public final String no_task_message = line + "\n" + indentation + "Sorry, you do not have this task" + "\n" + line;
+    public final String task_num_message = line + "\n" + indentation + "Sorry, please enter a task number bigger than 0" + "\n" + line;
+    public final String lack_content_message = line + "\n" + indentation + "OOPS!!! The description or time of a task cannot be empty." + "\n" + line;
```
For the constant names maybe we need to use uppercase and underscore. But here the attributes are `final`, not `final static`. Maybe we need to double-check the naming rules.

Li-Lehao commented on src/main/java/duke/core/UI.java:
```diff
@@ -0,0 +1,28 @@
+    public final String added_message = line + "\n" + indentation + "Got it. I've added this task:" + "\n";
+    public final String reading_file_message = indentation + "Reading your tasks from record : )";
+    public final String creating_file_message = indentation + "Creating a file to store your tasks : )";
+    public final String wrong_message = index_message + "OOPS!! Something wrong!";
```
I really like the way you use `indentation` + `message` + `line`. It's very clear and nice.

Li-Lehao commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,58 @@
+        }
+        return false;
+    }
+
```
The function name `searchKeyword` should be changed to `hasKeyword` to make it more like a boolean. The way you match keywords is very concise and clear. Would you consider making it case-insensitive?

Leofeng10 committed:
Ui class: Strings are very long and there are all in one line
To improve readability and code quality
Split the code into multiple lines
This is compliant with java code standard

Leofeng10 committed:
Assertion

Leofeng10 committed:
Merge pull request #1 from Leofeng10/branch-A-Assertions

Assertion

Leofeng10 committed:
improve code quility

Leofeng10 committed:
Merge pull request #2 from Leofeng10/branch-A-CodeQuality

improve code quility

Leofeng10 committed:
Stream

Leofeng10 committed:
Merge pull request #3 from Leofeng10/branch-A-Streams

Stream

Leofeng10 committed:
add Continuous integration

Leofeng10 committed:
Add recurring task

Leofeng10 committed:
A-BetterGui

Leofeng10 committed:
A-UserGuide

Leofeng10 committed:
Updat UG

Leofeng10 committed:
bug fixed

Leofeng10 committed:
bug fixed

Leofeng10 committed:
update user guide

Leofeng10 committed:
Update README.md

Leofeng10 committed:
Update README.md

Leofeng10 committed:
update readme

Leofeng10 committed:
Create Test for different classes

Leofeng10 committed:
Merge branch 'master' of https://github.com/Leofeng10/ip

Leofeng10 committed:
check style update

Leofeng10 committed:
coding standard

Leofeng10 committed:
coding standard

Leofeng10 committed:
coding standard

Leofeng10 committed:
coding standard

Leofeng10 committed:
update

Leofeng10 committed:
coding standard

Leofeng10 committed:
gradle

Leofeng10 committed:
coding standard

Leofeng10 committed:
coding standard

Leofeng10 committed:
coding standard

Leofeng10 committed:
update ui

Leofeng10 committed:
coding standard

Leofeng10 committed:
comment on method

Leofeng10 committed:
bug fixed
